src/assertions.cpp

```diff
@@ -206,6 +206,14 @@ class AssertExtractor {
                     Assert{TypeConstraint{ins.dst, TypeGroup::ptr_or_num}}
                 };
             }
+        case Bin::Op::DIV:
+        case Bin::Op::MOD:
+            if (std::holds_alternative<Reg>(ins.v)) {
+                auto src = reg(ins.v);
+                return {Assert{TypeConstraint{ins.dst, TypeGroup::number}}, Assert{ValidDivisor{src}}};
```

Comment:
It's possible to use `allow_division_by_zero` here, which saves some computation and clutter.

Reply:
ebpf_domain.cpp currently also checks that ValidDivisor is a number which would apply even if allow_division_by_zero is set.  Maybe there's some other code path that already does that though, I didn't add a test for that.